### PR TITLE
[IMPROVEMENT] Revamped Toast in place of EventEmitter

### LINKED ISSUE #2536: Export show toast function
### Current way
Out [Toast component](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/containers/Toast.js) works based on EventEmitter and [every time we need to call it, we emit an event](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/views/StatusView.js#L119).

### Proposed way
It'd be easier if we exported a `showFunction` receiving the message.

### PR DESCRIPTION
**Pull request by aKn1ghtOut**
## Proposed changes

Removed implementation of the toast functionality using EventEmitters. Instead, transferred the functionality to a react hook with a provider.

## Issue(s)	

Closes #2536 

## How to test or reproduce

1. Touch and hold down on any message
2. Select permalink
3. The toast works

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/containers/Toast.js [hunk 1]
```diff
@@ -20,15 +19,17 @@ const styles = StyleSheet.create({
...
 });
 
-export const LISTENER = 'Toast';
-
 class Toast extends React.Component {
 	static propTypes = {
 		theme: PropTypes.string
 	}
 
+	static mounted = null;
```
Comment (diegolmello) [inline app/containers/Toast.js:27]: Do we really need `mounted` here?

File: app/containers/Toast.js [hunk 2]
```diff
@@ -20,15 +19,17 @@ const styles = StyleSheet.create({
...
+
+	static show = (...args) => Toast.mounted && Toast.mounted.showToast(...args);
```
Comment (diegolmello) [inline app/containers/Toast.js:29]: It's really odd how `showToast` is a child of `mounted`.

Comment (diegolmello) [inline app/containers/Toast.js:29]: Instead of static, can we use refs instead? Maybe even migrate to hooks.

**Review by diegolmello — CHANGES_REQUESTED** [review]
@aKn1ghtOut Thanks.
Left a few comments.
I also wonder if we have a better option by this point.
This lib is really easy to use, but it also isn't testable on Detox.

File: app/containers/Toast.js (hunk 2 above)
Comment (aKn1ghtOut) [reply app/containers/Toast.js:29]: I do agree that it's odd, yeah

Comment (aKn1ghtOut) [reply app/containers/Toast.js:29]: Looking into it!

**Comment by aKn1ghtOut** [comment]
@diegolmello Switched the implementation to use hooks. Feels much better. Please review.

**Pull request closed without merging**